src/App.js

```diff
@@ -1,50 +1,68 @@
 import React from "react";
 import './App.css';
 
-import TypoGraphy from '@material-ui/core/Typography'
 
-import {Login} from "./Components/Login/Login";
-import {Register} from "./Components/Register/Register";
-import {BrowserRouter as Router, Route, Switch, Redirect} from "react-router-dom";
-import Dashboard from "./Components/Dashboard/Dashboard";
-import Header from "./Components/Header/Header";
+import {BrowserRouter as Router, Route, Switch, Redirect, withRouter} from "react-router-dom";
+
+
 import Footer from "./Components/Footer/Footer";
+import RegisterContainer from "./Components/Register/RegisterContainer";
+import LoginContainer from "./Components/Login/LoginContainer";
+import store from "./redux/redux-store";
+import DashboardContainer from "./Components/Dashboard/DashboardContainer";
+import HeaderContainer from "./Components/Header/HeaderContainer";
+import {connect} from "react-redux";
+import {getAuthUserData} from "./redux/auth-reducer";
+import {compose} from "redux";
+
+
 
+class App extends React.Component {
 
-function App() {
-    const auth = localStorage.getItem('AUTH');
+    constructor(props) {
+        super(props);
 
-    return (
-        <div className='app-wrapper'>
 
+        this.state = {
+            isAuth: store.getState().auth.isAuth || null
```

Comment:
I suggest to use `mapStateToProps` and `connect` instead direct access to the store

Reply:
And this data is already in the redux store, you don't need to duplicate it inside compoentn

Reply:
Simple variant how to improve:

class App extends React.Component {
    render() {
        const {isAuth} = this.props;
        return (
            <Router>
                <div className='app-wrapper'>
                    <HeaderContainer/>

                    {!isAuth && <Switch>
                        <Route path='/login' component={LoginContainer}/>
                        <Route path="/register" component={RegisterContainer}/>
                        <Route path="/dashboard/:boardId?" component={DashboardContainer}/>
                        <Redirect to="/login"/>
                    </Switch>}

                    {!!isAuth && <Switch></Switch>}
                    <Footer/>
                </div>
            </Router>
        )

    }
}

const mapStateToProps = state => ({
    isAuth: !!state.auth.accessToken
});

export default compose(
    withRouter,
    connect(mapStateToProps, getAuthUserData()))(App);